source/content/advanced-redirects.md

```diff
@@ -0,0 +1,334 @@
+      newrelic_name_transaction("redirect");
+    }
+
+    exit();
+  }
+
+}
+```
+
+## Redirect Legacy UNIX-Style User Home Folder Paths
+
+When transitioning from a system that used a tilde to indicate a home directory, the syntax is slightly different. Here's how you can parse out the username and relative path that the request was made for:
+
+```php:title=wp-config.php%20or%20settings.php
+$request_parts = explode('/', $_SERVER['REQUEST_URI']);
+$legacy_username = $legacy_path = '';
+if (isset($request_parts[1]) && strpos($request_parts[1], '~') === 0) {
+  $legacy_username = substr($request_parts[1], 1);
+  // If FALSE, then the request was just to the username.
+  $legacy_path = substr($_SERVER['REQUEST_URI'], (strlen($request_parts[1]) + 1));
+}
+if ($legacy_username) {
+  // Your custom logic.
+}
+```
+
+## Redirect to Force Lowercase Letters
+
+<TabList>
+
+<Tab title="WordPress" id="wp-lc" active={true}>
+
+WordPress automatically forces lowercase letters within URLs using the [`sanitize_title_with_dashes()`](https://core.trac.wordpress.org/browser/tags/4.6/src/wp-includes/formatting.php#L1744) function in core.
+
+</Tab>
+
+<Tab title="Drupal" id="dp-lc">
+
+Drupal sites can force lowercase letters using the following:
+
+1. Set general automatic alias settings to **Change to lower case** within the [PathAuto](https://www.drupal.org/project/pathauto) module configuration (`/admin/build/path/pathauto`).
+
+1. Enable **Case Sensitive URL Checking** within the [Global Redirect](https://www.drupal.org/project/globalredirect) module configuration (`/admin/settings/globalredirect`).
+
+</Tab>
+
+</TabList>
+
+## Redirect Files
+
+Because Drupal or WordPress aren't bootstrapped when static assets (e.g, images, PDFs, HTML files) are served, the PHP redirects used above will not work when these files are requested directly. You can use [CloudFlare](/cloudflare) or another stacked CDN to handle file redirects.
+
+Alternatively, you can remove the file entirely from the old location. In this case, the request will run through Drupal or WordPress. You can let the CMS serve a 404, or you can utilize a redirect in `wp-config.php` or `settings.php` as shown in the examples above.
+
+## Restrict File Access Based on IP
+
+If you want to restrict access to files based on the source IP address of the request, you can do so with PHP.
+
+<TabList>
+
+<Tab title="WordPress" active={true} id="restrict-wp">
+
+The following example restricts access to `/wp-admin/` and `/wp-login.php` based on the IP addresses listed in the `trusted_ips` array:
+
+```php:title=wp-config.php
+function ip_in_list($ips) {
+    foreach(explode(',', $_SERVER['HTTP_X_FORWARDED_FOR']) as $check_ip) {
+        foreach($ips as $ip) {
+          if(FALSE !== strpos($check_ip, $ip)) {
+            return true;
+          }
+        }
+    }
+    return false;
+  }
+function is_from_trusted_ip() {
+   $trusted_ips = array('192.0.2.38','198.51.100.12','208.0.113.159','2001:DB8:1C93');
```

Comment:
Insert a comment above this line that instructs the reader to replace IPs

Reply:
Same feedback for drupal snippet

Reply:
Comments added in https://github.com/pantheon-systems/documentation/pull/5516/commits/a7e4c20988a819b75567c926d7678d71063e686a